src/sgame/sg_bot_util.cpp

```diff
@@ -2095,9 +2095,40 @@ void BotFireWeaponAI( gentity_t *self )
 			}
 			break;
 		case WP_LUCIFER_CANNON:
-			if ( self->client->ps.weaponCharge < LCANNON_CHARGE_TIME_MAX * Math::Clamp( random(), 0.5f, 1.0f ) )
 			{
-				BotFireWeapon( WPM_PRIMARY, botCmdBuffer );
+				const int CRITICAL_HEALTH = 30; //TODO do not use a constant
+				int selfHP = self->client->ps.stats[ STAT_HEALTH ];
+
+				gentity_t const* botTarget = self->botMind->goal.getTargetedEntity();
+				float targetHP;
```

Comment:
Why is this set to the max health rather than the current health?

Reply:
Because i do not like cheating bots. Using actual health means that even if they catch a fleeing enemy (without hearing it taking a shot) they'll still know exactly how much to load their weapon, which is basically cheating.

Reply:
It should be named `targetMaxHP` then.

So ignoring the self HP condition, this amounts to use secondary fire iff the target is a dretch. That seems quite different from the strategy most human players use which is to shoot the floor with the primary attack, targeting dretches with splash damage. Maybe it could work if the bot's aim is really good

Reply:
it's actually very good. And I think players use splash because this allows to hit stuff around, at the expense of requiring more time to load. Not sure. Myself I also use the 2ndary when fighting dretches... because it's faster to reuse, too.
I guess I can just make bots use primary all the time if you prefer